nav2_regulated_pure_pursuit_controller/src/collision_checker.cpp

```diff
@@ -90,16 +90,21 @@ bool CollisionChecker::isCollisionImminent(
   curr_pose.x = robot_pose.pose.position.x;
   curr_pose.y = robot_pose.pose.position.y;
   curr_pose.theta = tf2::getYaw(robot_pose.pose.orientation);
+  double accumulated_distance = 0.0;
 
   // only forward simulate within time requested
   int i = 1;
-  while (i * projection_time < params_->max_allowed_time_to_collision_up_to_carrot) {
+  while (i * projection_time < params_->max_allowed_time_to_collision_up_to_carrot ||
+    (accumulated_distance < params_->min_distance_to_obstacle &&
+    projection_time * linear_vel > 0.01))
```

Comment:
I suppose maybe I'm also missing the detail: do you want these both conditions to need to be satisfied to stop, or just one or other other?

If, for instance, you want both, which I think you do, we could instead do something like:

`max_allowed_time_to_collision_check = std::max(params_->max_allowed_time_to_collision_up_to_carrot, params_->min_distance_to_obstacle * linear_vel);` and use `max_allowed_time_to_collision_check` in the `while` loop condition instead of the parameter. I think that would make this intent more clear.